js/codemeta_generation.js

```diff
@@ -321,7 +323,11 @@ function importReview(doc) {
 }
 
 function authorsEqual(author1, author2) {
-    // TODO should test more properties for equality?
+    const id1 = typeof author1 === "string"? author1 : getDocumentId(author1);
+    const id2 = typeof author2 === "string"? author2 : getDocumentId(author2);
+    if (id1 || id2) {
+        return id1 === id2;
+    }
```

Comment:
```suggestion
    const id1 = typeof author1 === "string"? author1 : getDocumentId(author1);
    const id2 = typeof author2 === "string"? author2 : getDocumentId(author2);
    if (id1 || id2) {
        // Use their id if both authors have one
        return id1 === id2;
    }
    // Fall back to comparing values otherwise
```